File: Tool/PreUtil/src/output/javascript/medium/DeepestLeavesSum.js
/**https://leetcode.com/problems/deepest-leaves-sum/ */
//Given the root of a binary tree, return the sum of values of its deepest leaves.


/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var deepestLeavesSum = function(root) {
    let maxDepth = 0;
    let sum = 0;
    let dfs = function(node, depth){
        if(node == null){
            return;
        }
        if(node.left == null && node.right == null){
            if(depth > maxDepth){
                maxDepth = depth;
                sum = node.val;
            }else if(depth == maxDepth){
                sum += node.val;
            }
        }
        dfs(node.left, depth + 1);
        dfs(node.right, depth + 1);
    }
    dfs(root, 0);
    return sum;
    
};

    

File: Tool/PreUtil/src/output/c/medium/DeepestLeavesSum.c
/**https://leetcode.com/problems/deepest-leaves-sum/ */
//Given the root of a binary tree, return the sum of values of its deepest leaves.


/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int deepestLeavesSum(struct TreeNode* root){
    int result = 0;
    int maxDepth = 0;
    struct TreeNode* queue[10000];
    int queueSize = 0;
    queue[queueSize++] = root;
    while(queueSize > 0){
        int size = queueSize;
        queueSize = 0;
        result = 0;
        maxDepth++;
        for(int i = 0; i < size; i++){
            struct TreeNode* node = queue[i];
            result += node->val;
            if(node->left != NULL){
                queue[queueSize++] = node->left;
            }
            if(node->right != NULL){
                queue[queueSize++] = node->right;
            }
        }
    }
    return result;

}

    